src/ci/docker/host-x86_64/dist-various-1/install-riscv32-none-elf.sh

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+set -ex
+
+curl -L https://github.com/riscv-collab/riscv-gnu-toolchain/releases/download/2023.10.18/riscv32-elf-ubuntu-22.04-gcc-nightly-2023.10.18-nightly.tar.gz \
```

Comment:
We will probably want to mirror these tarballs into our s3 bucket (GitHub download urls can be unreliable).

However I want to check - is there not gcc riscv compilers in Ubuntu's standard toolchains? Or perhaps we could use e.g. clang here?

Reply:
> We will probably want to mirror these tarballs into our s3 bucket (GitHub download urls can be unreliable).

I assume that isn't something I do as part of this PR. Let me know if I need to do anything though

> However I want to check - is there not gcc riscv compilers in Ubuntu's standard toolchains? Or perhaps we could use e.g. clang here?

There is one for [64-bit](https://packages.ubuntu.com/mantic/gcc-riscv64-unknown-elf) but AFAIK there aren't any for 32-bit. The version also seems a little out of date, I vaguely think GCC 10 has some RISC-V bugs, but I'm not sure if that impacts us here.

As RISC-V support improves we can look at removing these and using versions shipped by distros.

Reply:
I see, okay. I'll likely get a chance tomorrow to upload/mirror these to our bucket, and post the resulting URLs here for an update. We can merge after that.

Reply:
Alright, https://ci-mirrors.rust-lang.org/rustc/riscv32-elf-ubuntu-22.04-gcc-nightly-2023.10.18-nightly.tar.gz  and https://ci-mirrors.rust-lang.org/rustc/riscv64-elf-ubuntu-22.04-gcc-nightly-2023.10.18-nightly.tar.gz should work. Please update the PR with those, squash commits, and then I'm happy to merge.

Reply:
Done!